### PR TITLE
#139 Compact mode of file comments

### PR DESCRIPTION
**Pull request by bmaehr**
Compact mode of file comments

### CODE REVIEW

File: src/main/java/com/github/mc1arke/sonarqube/plugin/CommunityBranchPlugin.java [hunk 1]
```diff
@@ -179,7 +179,11 @@ public void load(CoreExtension.Context context) {
                             .name("Repository Slug for the Gitlab (Server or Cloud) instance")
                             .description("The repository slug can be either in the form of user/repo or it can be the Project ID")
                             .type(PropertyType.STRING)
-                            .build()
+                            .build(),
+                            
+                    PropertyDefinition.builder(GitlabServerPullRequestDecorator.PULLREQUEST_COMPACT_COMMENTS_ENABLED).category(PULL_REQUEST_CATEGORY_LABEL).subCategory(GITLAB_INTEGRATION_SUBCATEGORY_LABEL)
```
Comment (4n4n4s) [inline src/main/java/com/github/mc1arke/sonarqube/plugin/CommunityBranchPlugin.java]: this would be for me also a property that can be used by the bitbucket provider too. I see you hardcoded it currenlty there to false.

Ideally the property is available for all providers.

File: src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabServerPullRequestDecoratorTest.java [hunk 2]
```diff
@@ -66,7 +66,8 @@ public void decorateQualityGateStatus() {
         when(configuration.get(PullRequestBuildStatusDecorator.PULL_REQUEST_COMMENT_SUMMARY_ENABLED)).thenReturn(Optional.of("true"));
         when(configuration.get(PullRequestBuildStatusDecorator.PULL_REQUEST_DELETE_COMMENTS_ENABLED)).thenReturn(Optional.of("true"));
         when(configuration.get(PullRequestBuildStatusDecorator.PULL_REQUEST_FILE_COMMENT_ENABLED)).thenReturn(Optional.of("true"));
-
+        when(configuration.get(GitlabServerPullRequestDecorator.PULL_REQUEST_COMPACT_COMMENTS_ENABLED)).thenReturn(Optional.of("true"));
+
```
Comment (mc1arke) [inline src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabServerPullRequestDecoratorTest.java:70]: Please add a test case for the negative version of this condition

**Comment by mc1arke** [comment]
@bmaehr are you still interested in progressing this feature?
